app.ts

```diff
@@ -0,0 +1,61 @@
+/**  Debounce utility function to limit function execution frequency*/
+function debounce<F extends (...args: any) => any>(func: F, waitFor: number) {
+	let timeout: number;
+
+	return (...args: Parameters<F>): Promise<ReturnType<F>> => {
+		clearTimeout(timeout);
+
+		return new Promise((resolve) => {
+			timeout = setTimeout(() => {
+				resolve(func(...args));
+			}, waitFor);
+		});
+	};
+}
+
+/**  Constants for grid calculation
+ * GRID_RATIO represents the ratio of the grid's height to the window's height.
+*/
+const GRID_RATIO = 9/20 ;
+const ROW_HEIGHT = 16;
+
+/**  Wait for the document to be ready*/
```

Comment:
One space to many at the start of this comment, and please add a space before the `*/`